sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java

```diff
@@ -123,7 +127,9 @@
     }
 
     sqlEnvBuilder.setQueryPlannerClassName(
-        input.getPipeline().getOptions().as(BeamSqlPipelineOptions.class).getPlannerName());
+        MoreObjects.firstNonNull(
+            queryPlannerClassName(),
+            input.getPipeline().getOptions().as(BeamSqlPipelineOptions.class).getPlannerName()));
```

Comment:
Do you think if it is helpful to find a "working" class name, rather than a "non null" parameter? (meaning that try first class name even if it is non-null, if there is an exception, try the next one.)

Reply:
Hm.. I think maybe that would be surprising for the user. 
I think it would be better to just try to use a non-working class name and fail, than to fall back to a different planner - since changing the planner has implications on acceptable query syntax.

Reply:
that makes sense